libtcmu.c

```diff
@@ -325,6 +343,58 @@ static void close_devices(struct tcmulib_context *ctx)
 	}
 }
 
+static int reconfig_device(struct tcmulib_context *ctx, char *dev_name, char *cfgstring,
+			   uint32_t cfgtype)
+{
+	struct tcmu_device **dev_ptr;
+	struct tcmu_device *dev;
+	int i = 0;
+	bool found = false;
+	int ret;
+
+	if (cfgtype == NO_RECONFIG)
+		return 0;
+
+	darray_foreach(dev_ptr, ctx->devices) {
+		dev = *dev_ptr;
+		size_t len = strnlen(dev->dev_name, sizeof(dev->dev_name));
+		if (strncmp(dev->dev_name, dev_name, len)) {
+			i++;
+		} else {
+			found = true;
+			break;
+		}
+	}
+
+	if (!found) {
+		tcmu_dev_err(dev, "could not reconfigure device %s: not found\n",
+			     dev_name);
+		goto err_free;
```

Comment:
You are not freeing anything so probably should modify the label to reflect that, or just drop the gotos and do returns like you do a couple times below. Just sync up on one or the other. Don't mix.